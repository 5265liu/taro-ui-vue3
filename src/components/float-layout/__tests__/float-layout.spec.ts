import { mount } from '@vue/test-utils'
import AtFloatLayout from '../index'
import { sleep } from '@/tests/helper'

const factory = (
  values = {},
  slots = {
    default: [
      `这是内容区 随你怎么写这是内容区 随你怎么写这是内容区
随你怎么写这是内容区 随你怎么写这是内容区 随你怎么写这是内容区
随你怎么写`,
    ],
  }
) => {
  return mount(AtFloatLayout as any, {
    global: {
      components: {}
    },
    slots,
    props: { ...values },
  })
}

describe('FloatLayout Snap', () => {
  it('render initial FloatLayout', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render FloatLayout -- props isOpened', () => {
    const wrapper = factory({ isOpened: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render FloatLayout -- props title', () => {
    const wrapper = factory({ title: '这是个标题', isOpened: true })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('FloatLayout Behavior ', () => {
  it('FloatLayout onClose', async () => {
    const onClose = jest.fn()
    const wrapper = factory({
      title: '这是个标题',
      isOpened: true,
      onClose: onClose,
    })
    wrapper.find('.at-float-layout__overlay').trigger('tap')
    await sleep(0)
    expect(onClose).toBeCalled()
  })
})
