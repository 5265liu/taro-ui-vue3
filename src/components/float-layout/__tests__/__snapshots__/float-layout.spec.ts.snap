// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FloatLayout Snap render FloatLayout -- props isOpened 1`] = `
<view
  class="at-float-layout at-float-layout--active"
>
  <view
    class="at-float-layout__overlay"
  />
  <view
    class="at-float-layout__container layout"
  >
    
    <view
      class="layout-body"
    >
      <scroll-view
        class="layout-body__content"
        scrollwithanimation="false"
        scrollx="false"
        scrolly="true"
      >
        这是内容区 随你怎么写这是内容区 随你怎么写这是内容区
随你怎么写这是内容区 随你怎么写这是内容区 随你怎么写这是内容区
随你怎么写
      </scroll-view>
    </view>
  </view>
</view>
`;

exports[`FloatLayout Snap render FloatLayout -- props title 1`] = `
<view
  class="at-float-layout at-float-layout--active"
>
  <view
    class="at-float-layout__overlay"
  />
  <view
    class="at-float-layout__container layout"
  >
    <view
      class="layout-header"
    >
      <text
        class="layout-header__title"
      >
        这是个标题
      </text>
      <view
        class="layout-header__btn-close"
      />
    </view>
    <view
      class="layout-body"
    >
      <scroll-view
        class="layout-body__content"
        scrollwithanimation="false"
        scrollx="false"
        scrolly="true"
      >
        这是内容区 随你怎么写这是内容区 随你怎么写这是内容区
随你怎么写这是内容区 随你怎么写这是内容区 随你怎么写这是内容区
随你怎么写
      </scroll-view>
    </view>
  </view>
</view>
`;

exports[`FloatLayout Snap render initial FloatLayout 1`] = `
<view
  class="at-float-layout"
>
  <view
    class="at-float-layout__overlay"
  />
  <view
    class="at-float-layout__container layout"
  >
    
    <view
      class="layout-body"
    >
      <scroll-view
        class="layout-body__content"
        scrollwithanimation="false"
        scrollx="false"
        scrolly="true"
      >
        这是内容区 随你怎么写这是内容区 随你怎么写这是内容区
随你怎么写这是内容区 随你怎么写这是内容区 随你怎么写这是内容区
随你怎么写
      </scroll-view>
    </view>
  </view>
</view>
`;
